lib/logstash/outputs/sqs.rb

```diff
@@ -98,9 +101,13 @@ def register
    end
 
     begin
-      @logger.debug('Connecting to SQS queue', :queue => @queue, :region => region)
-      @queue_url = @sqs.get_queue_url(:queue_name => @queue)[:queue_url]
-      @logger.info('Connected to SQS queue successfully', :queue => @queue, :region => region)
+      @logger.debug('Connecting to SQS queue', :queue => @queue, :region => region, :queue_owner_aws_account_id => @queue_owner_aws_account_id)
```

Comment:
It might be nice if the `queue_owner_aws_account_id` property only shows up if it has been explicitly set - otherwise this shows up in the logs as `queue_owner_aws_account_id => nil`

Reply:
I suppose that would require either moving the logging inside the if statement:
```ruby
if @queue_owner_aws_account_id
  @logger.debug('Connecting to cross-account SQS queue', :queue => @queue, :region => region, :queue_owner_aws_account_id => @queue_owner_aws_account_id)
  @sqs.get_queue_url(:queue_name => @queue, :queue_owner_aws_account_id => @queue_owner_aws_account_id)[:queue_url]
  @logger.info('Connected to cross-account SQS queue successfully', :queue => @queue, :region => region, :queue_owner_aws_account_id => @queue_owner_aws_account_id)
else
  @logger.debug('Connecting to SQS queue', :queue => @queue, :region => region)
  @sqs.get_queue_url(:queue_name => @queue)[:queue_url]
  @logger.info('Connected to SQS queue successfully', :queue => @queue, :region => region)
end
```
which seems a bit duplicative.

Or perhaps some form of hash manipulation:
```ruby
queue_merger = @queue_owner_aws_account_id ? {:queue_owner_aws_account_id => @queue_owner_aws_account_id} : {}
@logger.debug('Connecting to SQS queue', {:queue => @queue, :region => region}.merge(queue_merger))
@queue_url = @sqs.get_queue_url({:queue_name => @queue}.merge(queue_merger))[:queue_url]
@logger.info('Connected to SQS queue successfully', {:queue => @queue, :region => region}.merge(queue_merger))
```
which is convoluted and messy.

Maybe a hybrid approach?
```ruby
logger_params = { :queue => @queue, :region => region }
logger_params[:queue_owner_aws_account_id] = @queue_owner_aws_account_id if @queue_owner_aws_account_id
@logger.debug('Connecting to SQS queue', logger_params)
@queue_url = if @queue_owner_aws_account_id
  @sqs.get_queue_url(:queue_name => @queue, :queue_owner_aws_account_id => @queue_owner_aws_account_id)[:queue_url]
else
  @sqs.get_queue_url(:queue_name => @queue)[:queue_url]
end
@logger.info('Connected to SQS queue successfully', logger_params)
```

Any which way you slice it, it seems unnecessarily messy in the code for the value added in the logs.

Reply:
Since `:region` isn't passed to `qs.get_queue_url` couldn't @Cobi 's hybrid approach be used, but with passing the prepared hash into all called methods?
```ruby
params = { queue: @queue }
params[:queue_owner_aws_account_id] = @queue_owner_aws_account_id if @queue_owner_aws_account_id
@logger.debug('Connecting to SQS queue', params)
@queue_url = @sqs.get_queue_url(params)
@logger.info('Connected to SQS queue successfully', params)
```